### PR TITLE
added comment lines for code review

### PR DESCRIPTION
**Pull request by athackst**
Trying out different code review sites

---

This change is [Reviewable](https://reviewable.io/reviews/toyotaresearchinstitute/cartographer_hsr/1)

### CODE REVIEW

File: cartographer_hsr.rosinstall [hunk 1]
```diff
@@ -1,3 +1,4 @@
 - git: {local-name: src/cartographer, uri: 'https://github.com/googlecartographer/cartographer.git'}
 - git: {local-name: src/cartographer_ros, uri: 'https://github.com/googlecartographer/cartographer_ros.git'}
 - git: {local-name: src/cartographer_hsr, uri: 'https://github.com/ToyotaResearchInstitute/cartographer_hsr.git'}
+# comment added for code review
```
Comment (macmason-tri) [inline cartographer_hsr.rosinstall]: Newline between entry and comment line, please.

**Review by macmason-tri — CHANGES_REQUESTED** [review]
LGTM with nitpicks.

**Comment by athackst** [comment]
BTW The auto-genned format for a rosinstall has a slightly different but just-as-valid yaml format like the following:

```
- git:
    local-name: <name>
    uri: <uri>
```

I slightly prefer this formatting since it allows me to copy and paste other rosinstall entries into my .rosinstall file and have it match.

---

Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

_[cartographer_hsr.rosinstall, line 1 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNVRFNL45E_fz18DFT:-KSNUvwS2mpHJP5hxyHf:b77j6vp) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L1)):_

> ```
> - git: {local-name: src/cartographer, uri: 'https://github.com/googlecartographer/cartographer.git'}
> ```

cartographer actually takes quite more than this to setup (https://google-cartographer.readthedocs.io/en/latest/), so it's not really appropriate to include it in the rosinstall file like this.  You can, however, include a setup script (http://docs.ros.org/independent/api/rosinstall/html/rosinstall_file_format.html) if you wish.  Otherwise it's probably best to just point to the setup page in the README.

---

_[cartographer_hsr.rosinstall, line 2 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNWWrgtpqxVj9C7Zh5:-KSNWWrgtpqxVj9C7Zh6:b-137xra) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L2)):_

> - git: {local-name: src/cartographer, uri: 'https://github.com/googlecartographer/cartographer.git'}
> - git: {local-name: src/cartographer_ros, uri: 'https://github.com/googlecartographer/cartographer_ros.git'}
> - git: {local-name: src/cartographer_hsr, uri: 'https://github.com/ToyotaResearchInstitute/cartographer_hsr.git'}

Please remove the "src" from the local-name.  rosinstalls are already placed within the "src" directory and this creates a nested src folder.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-:-KSNYwIbapicYIr1uNvX:bv781hb)_

**Comment by macmason-tri** [comment]
Followed the cartographer convention here, but I have no strong feelings: let me know what you'd like best.

---

Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

_[cartographer_hsr.rosinstall, line 1 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNVRFNL45E_fz18DFT:-KSNw8tLzGqkMOIldW_O:bp5ohxg) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L1)):_
<details><summary>

<i>Previously, athackst (Allison Thackston) wrote…</i></summary>



> cartographer actually takes quite more than this to setup (https://google-cartographer.readthedocs.io/en/latest/), so it's not really appropriate to include it in the rosinstall file like this.  You can, however, include a setup script (http://docs.ros.org/independent/api/rosinstall/html/rosinstall_file_format.html) if you wish.  Otherwise it's probably best to just point to the setup page in the README.
> </details>
> ...how many different package manager models does ROS use at the same time? Eleven?

I don't think it actually requires any extra setup. It's the usual pattern: wstool merge, rosdep install, catkin_make.  I've added some text to README.rst explaining what to do.

---

_[cartographer_hsr.rosinstall, line 2 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNWWrgtpqxVj9C7Zh5:-KSO-XqmWG_zc-naq1ig:bmxbn5p) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L2)):_
<details><summary>

<i>Previously, athackst (Allison Thackston) wrote…</i></summary>



> > - git: {local-name: src/cartographer, uri: 'https://github.com/googlecartographer/cartographer.git'}
> > - git: {local-name: src/cartographer_ros, uri: 'https://github.com/googlecartographer/cartographer_ros.git'}
> > - git: {local-name: src/cartographer_hsr, uri: 'https://github.com/ToyotaResearchInstitute/cartographer_hsr.git'}
> 
> Please remove the "src" from the local-name.  rosinstalls are already placed within the "src" directory and this creates a nested src folder.
> 
> </details>
> It doesn't add an extra src/ for me. New feature?

I'm running wstool as suggested by the Cartographer docs.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-:-KSNvhen8k2RZP1b5X7u:b-vqxgl1)_

**Comment by macmason-tri** [comment]
Reviewed 2 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1)_

**Comment by macmason-tri** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1)_

**Comment by athackst** [comment]
Reviewed 1 of 2 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

_[cartographer_hsr.rosinstall, line 1 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNVRFNL45E_fz18DFT:-KSO5RDapYzD0HMnMOH8:b-pf6vac) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L1)):_
Ha! at least one million.

Unfortunately hsr doesn't have instructions for custom rosdep keys, so that part won't work for us.

> ERROR: the following packages/stacks could not have their rosdep keys resolved
> to system dependencies:
> cartographer_hsr: Cannot locate rosdep definition for [hsrb_robot_launch]

(I also have an admittedly irrational pet-peeve against catkin_make_isolated for my main dev directory and typically install packages that don't play nice separately, but this isn't by any means a requirement)

---

_[cartographer_hsr.rosinstall, line 2 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNWWrgtpqxVj9C7Zh5:-KSO8NEPvnNzzrbu72UY:b-57qcfx) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L2)):_

> wstool merge https://raw.githubusercontent.com/ToyotaResearchInstitute/cartographer_hsr/master/cartographer_hsr.rosinstall

Which is different from the way that ROS suggests you do it http://wiki.ros.org/wstool

The ROS way is how I've seen most ROS developers do it, where the workspace is initialized in the src folder and not in the main folder.  Adding the additional "src" is both more verbose (why add the same string to every single package ever?) and breaks current ROS standards.  

---

_[cartographer_hsr.rosinstall, line 4 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNUbU2INlUcb1T0wgt-r1-4:-KSOH4TQ8RW1kfFICUwz:bew6iqb) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L4)):_
<details><summary>

<i>Previously, macmason-tri wrote…</i></summary>



> Newline between entry and comment line, please.
> </details>
> How about we just delete this line after the review is over instead?

---

_[CMakeLists.txt, line 20 at r3](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSOICSmzmJK4hkH1Y9X:-KSOICSmzmJK4hkH1Y9Y:b-vbaacu) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/42560eebde32fdd8a2a6dc170d27f89652118cd9/CMakeLists.txt#L20)):_

>  
> install(DIRECTORY launch/
>   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/launch/
> )
>  
> install(DIRECTORY configuration_files/
>   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/configuration_files/
> )

Fun sidebar: This is technically correct and does what you want, however it is not necessary to be so verbose and might actually be less error prone to do this the following way:

```
install(DIRECTORY launch
                                configuration_files
            DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
)
```

by omitting the trailing `\` the folder names will be created in `${CATKIN_PACKAGE_SHARE_DESTINATION}` and will have the same name in your working tree (which is desired because of how ROS handles the devel environment).

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1)_

**Comment by macmason-tri** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

_[cartographer_hsr.rosinstall, line 1 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNVRFNL45E_fz18DFT:-KSOa7QM-S3g9bOWcyz-:b-epy9ls) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L1)):_
<details><summary>

<i>Previously, athackst (Allison Thackston) wrote…</i></summary>



> Ha! at least one million.
> 
> Unfortunately hsr doesn't have instructions for custom rosdep keys, so that part won't work for us.
> 
> > ERROR: the following packages/stacks could not have their rosdep keys resolved
> > to system dependencies:
> > cartographer_hsr: Cannot locate rosdep definition for [hsrb_robot_launch]
> 
> (I also have an admittedly irrational pet-peeve against catkin_make_isolated for my main dev directory and typically install packages that don't play nice separately, but this isn't by any means a requirement)
> </details>
> So you're well into details I (a) never learned (viva rosmake!) or (b) don't remember (what's make_isolated do, again?). 

What's the right move here? Fix README.rst so it doesn't encourage rosdep? Suggest people just install cartographer, and then add this in a second pass?

---

_[cartographer_hsr.rosinstall, line 2 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNWWrgtpqxVj9C7Zh5:-KSOass_1pVdhmguhSA_:b8sgroo) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L2)):_
<details><summary>

<i>Previously, athackst (Allison Thackston) wrote…</i></summary>



> > wstool merge https://raw.githubusercontent.com/ToyotaResearchInstitute/cartographer_hsr/master/cartographer_hsr.rosinstall
> 
> Which is different from the way that ROS suggests you do it http://wiki.ros.org/wstool
> 
> The ROS way is how I've seen most ROS developers do it, where the workspace is initialized in the src folder and not in the main folder.  Adding the additional "src" is both more verbose (why add the same string to every single package ever?) and breaks current ROS standards.  
> 
> </details>
> Oh, right. This stupid catkin screwup.

("The directory is called 'workspace', but we "initialize" it in a subdirectory, for reasons...)

Fixed.

---

_[cartographer_hsr.rosinstall, line 4 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNUbU2INlUcb1T0wgt-r1-4:-KSObKgqkHPCuiUJvptz:b6maapv) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L4)):_
<details><summary>

<i>Previously, athackst (Allison Thackston) wrote…</i></summary>



> How about we just delete this line after the review is over instead?
> </details>
> Better yet, deleting it here!

---

_[CMakeLists.txt, line 20 at r3](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSOICSmzmJK4hkH1Y9X:-KSOb_QbWDa8Uqu5WX7p:b-sk4tqm) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/42560eebde32fdd8a2a6dc170d27f89652118cd9/CMakeLists.txt#L20)):_
<details><summary>

<i>Previously, athackst (Allison Thackston) wrote…</i></summary>



> >  
> > install(DIRECTORY launch/
> >   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/launch/
> > )
> >  
> > install(DIRECTORY configuration_files/
> >   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/configuration_files/
> > )
> 
> Fun sidebar: This is technically correct and does what you want, however it is not necessary to be so verbose and might actually be less error prone to do this the following way:
> 
> ```
> install(DIRECTORY launch
>                                 configuration_files
>             DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
> )
> ```
> 
> by omitting the trailing `\` the folder names will be created in `${CATKIN_PACKAGE_SHARE_DESTINATION}` and will have the same name in your working tree (which is desired because of how ROS handles the devel environment).
> </details>
> #include <what-mac-has-to-say-about-cmake>

PTAL.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1)_

**Comment by athackst** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r3, 2 of 3 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

_[cartographer_hsr.rosinstall, line 1 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNVRFNL45E_fz18DFT:-KSXHJsdPdT89_PE0reB:bjoeq1k) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L1)):_
<details><summary>

<i>Previously, macmason-tri wrote…</i></summary>



> So you're well into details I (a) never learned (viva rosmake!) or (b) don't remember (what's make_isolated do, again?). 
> 
> What's the right move here? Fix README.rst so it doesn't encourage rosdep? Suggest people just install cartographer, and then add this in a second pass?
> </details>
> Fix the README  (updated!) 

---

_[CMakeLists.txt, line 20 at r3](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSOICSmzmJK4hkH1Y9X:-KSOV-50buKqt8jVnimG:bb6ywmx) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/42560eebde32fdd8a2a6dc170d27f89652118cd9/CMakeLists.txt#L20)):_
<details><summary>

<i>Previously, macmason-tri wrote…</i></summary>



> #include <what-mac-has-to-say-about-cmake>
> 
> PTAL.
> </details>
> LGTM

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1)_

**Comment by macmason-tri** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-:-KSbHPDW0_p-Uc1StOor:bnfp4nl)_

**Comment by macmason-tri** [comment]
Reviewed 2 of 3 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1)_

**Review by macmason-tri — APPROVED** [review]

**Comment by macmason-tri** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[cartographer_hsr.rosinstall, line 4 at r1](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1#-KSNUbU2INlUcb1T0wgt-r1-4:-KSbZ0-sZdcu4GLYPm1c:bb70vlp) ([raw file](https://github.com/toyotaresearchinstitute/cartographer_hsr/blob/74485f52ddd826cdd67a20da377e5fa2689f115e/cartographer_hsr.rosinstall#L4)):_
<details><summary>

<i>Previously, macmason-tri wrote…</i></summary>



> Better yet, deleting it here!
> </details>
> OK.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/toyotaresearchinstitute/cartographer_hsr/1)_

**Pull request merged**